reana_job_controller/job_monitor.py

```diff
@@ -222,6 +223,9 @@ def get_job_logs(self, job_pod):
                         container.name, container.state.waiting.message
                     )
 
+            if job_pod.status.reason == "DeadlineExceeded":
+                pod_logs += "\nJob was killed due to timeout."
```

Comment:
*thought:* I am not sure this is a nice way/place to add additional logs. What if more reasons will be added? Was thinking to move it to a separate function like `enrich_logs_based_on_reason(job_pod, pod_logs) -> str`. Feedback and ideas are appreciated.

Reply:
Makes sense to me.

Reply:
Suggestion: What do you think about also logging (`logging.info()`) the same message (or similar) to the Job Controller for the sake of debuggability? If I'm not wrong, right now "Job was killed due to timeout." is only present in the workflow logs.

Reply:
I like the idea to put some logging. It could be very useful. I assume `logging.debug()` will not be enough.

Just thinking about where to put the logging code exactly... I can just add a line underneath this but not sure if it is the right place there. `get_job_logs` is responsible for getting pod's containers logs and not for checking status and adding some new logs to it.

Reply:
Come up with something like this. Not ideal but, at least, the logic that gets logs from containers is separated from logic that adds logs based on a pod reason. What do you think?

```python
# ...

def _get_containers_logs(job_pod) -> str:
    pod_logs = ""
    try:
        container_statuses = (job_pod.status.container_statuses or []) + (
            job_pod.status.init_container_statuses or []
        )

        logging.info(f"Grabbing pod {job_pod.metadata.name} logs ...")
        for container in container_statuses:
            if container.state.terminated:
                container_log = current_k8s_corev1_api_client.read_namespaced_pod_log(
                    namespace=REANA_RUNTIME_KUBERNETES_NAMESPACE,
                    name=job_pod.metadata.name,
                    container=container.name,
                )
                pod_logs += "{}: :\n {}\n".format(container.name, container_log)
                if hasattr(container.state.terminated, "reason"):
                    pod_logs += "\n{}\n".format(container.state.terminated.reason)
            elif container.state.waiting:
                pod_logs += "Container {} failed, error: {}".format(
                    container.name, container.state.waiting.message
                )
        return pod_logs
    except client.rest.ApiException as e:
        logging.error("Error while connecting to Kubernetes API: {}".format(e))
        return pod_logs
    except Exception as e:
        logging.error(traceback.format_exc())
        logging.error("Unexpected error: {}".format(e))
        return pod_logs


def get_job_logs(job_pod):
    """Get job logs."""
    logs = JobMonitorKubernetes._get_containers_logs(job_pod)

    if job_pod.status.reason == "DeadlineExceeded":
        logs += "\nJob was killed due to timeout."
        logging.info(f"Kubernetes job id: {job_pod.metadata.name} was killed due to timeout.")

    return logs
```